lib/vrl/stdlib/src/parse_xml.rs

```diff
@@ -300,12 +300,21 @@ fn process_node<'a>(node: Node, config: &ParseXmlConfig<'a>) -> Value {
     let recurse = |node: Node| -> BTreeMap<String, Value> {
         let mut map = BTreeMap::new();
 
+        // Expand attributes, if required.
+        if config.include_attr {
+            for attr in node.attributes() {
+                map.insert(
+                    format!("{}{}", config.attr_prefix, attr.name()),
+                    attr.value().into(),
+                );
+            }
+        }
+
         for n in node.children().into_iter().filter(|n| !n.is_comment()) {
-            // Use the default tag name if blank.
-            let name = if n.tag_name().name() == "" {
-                config.text_key.to_string()
-            } else {
-                n.tag_name().name().to_string()
+            let name = match n.node_type() {
+                NodeType::Element => n.tag_name().name().to_string(),
+                NodeType::Text => config.text_key.to_string(),
+                _ => unreachable!("shouldn't be other XML nodes"),
```

Comment:
This change is unrelated, I just found the existing code hard to reason about. I'm not crazy about the `unreachable` here but we were already doing that below. This function should only be called with elements or text nodes.